components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java

```diff
@@ -844,6 +848,53 @@ public Response getAttributes(@PathParam("siddhiAppName") String siddhiAppName,
                 ).build();
     }
 
+    /**
+     * Converts a given Siddhi App string to a specific JSON format for a graph that diagrammatically
+     * display's the Siddhi App to be generated in the Editor design view.
+     *
+     * @param siddhiAppBase64 The Siddhi App (encoded to Base64) to be converted to JSON
+     * @return The JSON result in a predefined format
+     */
+    @POST
+    @Path("/event-flow")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response constructEventFlowJsonString(String siddhiAppBase64) {
+        Response response;
+
+        try {
+            String siddhiAppString = new String(Base64.getDecoder().decode(siddhiAppBase64),
+                    Charset.defaultCharset());
```

Comment:
Why default charset?
default charset is platform and locale depend (https://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html#defaultCharset()), instead lets use `StandardCharsets.UTF_8 (https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8).

Reply:
Will make this change.